infrastructure/docker-compose.yml

```diff
@@ -1,85 +1,88 @@
-version: "3"
 services:
-  # NGINX reverse proxy server
   nginx:
-    image: nginx:1.18.0
+    image: nginx
     container_name: nginx
     restart: always
     networks:
       infranet:
         ipv4_address: 172.18.0.10
     ports:
-      - "80:80"
-      - "443:443"
+      - 80:80 # Web Interface
     hostname: "infra.corp.local"
     volumes:
-      - "./nginx/conf.d:/etc/nginx/conf.d"
-      - "./nginx/vhost.d:/etc/nginx/vhost.d"
-      - "./nginx/html:/usr/share/nginx/html"
-      - "./nginx/certs:/etc/nginx/certs:ro"
-      - "./nginx/proxy.conf:/etc/nginx/proxy.conf:ro"
-      - "/var/log/nginx:/var/log/nginx"
-  # GitLab CE Git repository manager
+      - "./nginx:/etc/nginx/"
+  nexus:
+    image: klo2k/nexus3 # Mac
+    #image: sonatype/nexus3 # Linux
+    container_name: nexus
+    hostname: "nexus.corp.local"
+    environment:
+      NEXUS_CONTEXT: nexus
+    networks:
+      infranet:
+        ipv4_address: 172.18.0.11
+    extra_hosts:
+      - "gitlab.corp.local:172.18.0.12"
+    ports:
+     - 8081:8081 # Web Interface
+    volumes:
+      - "nexus-data:/var/sonatype/work"
+    restart: always
+    ulimits:
+      nproc: 65535
+      nofile:
+        soft: 32000
+        hard: 40000
   gitlab:
-    image: gitlab/gitlab-ce:12.10.3-ce.0
+    image: yrzr/gitlab-ce-arm64v8 # Mac
+    #image: gitlab/gitlab-ce # Linux
     container_name: gitlab-ce
     restart: always
     environment:
       GITLAB_OMNIBUS_CONFIG: |
-        external_url "http://infra.corp.local/gitlab"
+        external_url 'http://infra.corp.local/gitlab'
     networks:
       infranet:
-        ipv4_address: 172.18.0.11
+        ipv4_address: 172.18.0.12
     extra_hosts:
-      - "infra.corp.local:<DOCKER_HOST_IP>"
+      - "infra.corp.local:172.18.0.10"
     ports:
-    - "127.0.0.1:8080:80"
+      - 8082:80 # Web Interface
```

Comment:
Interesting but for the gitlab-ce i got an error: "ERROR: for gitlab  Cannot create container for service gitlab: invalid port specification: "481342"
In order to fix it, I had to use quotes. I am not sure why it happened only for this container..
    ports:
      - "8082:80" # Web Interface
      - "8022:22" # SSH